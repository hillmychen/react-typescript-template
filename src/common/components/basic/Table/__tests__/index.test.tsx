import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import { Table } from '../index'

interface DataSourceItem {
  id: string
  name: string
}

describe('Table', () => {
  test('test UI', () => {
    const { asFragment, getByText } = render(
      <Table
        dataSource={[
          { id: '1', name: 'test1' },
          { id: '2', name: 'test2' },
          { id: '3', name: 'test3' },
          { id: '4', name: 'test4' },
          { id: '5', name: 'test5' },
          { id: '6', name: 'test6' },
          { id: '7', name: 'test7' },
          { id: '8', name: 'test8' },
          { id: '9', name: 'test9' },
          { id: '10', name: 'test10' },
          { id: '11', name: 'test11' },
        ]}
        pageSize={10}
        current={1}
        total={2}
      >
        <Table.Column<DataSourceItem> title="id" dataIndex="id" />
        <Table.Column<DataSourceItem> title="name" dataIndex="name" />
      </Table>
    )
    expect(getByText('test1')).toBeInTheDocument()
    expect(getByText('test2')).toBeInTheDocument()
    expect(getByText('共2条')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
