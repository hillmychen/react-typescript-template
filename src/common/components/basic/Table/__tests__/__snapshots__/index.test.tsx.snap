// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table test UI 1`] = `
<DocumentFragment>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-default ant-table-bordered ant-table-scroll-position-left"
        >
          <div
            class="ant-table-content"
          >
            <div
              class="ant-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col />
                  <col />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            id
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class="ant-table-row-cell-last"
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            name
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="0"
                  >
                    <td
                      class=""
                    >
                      1
                    </td>
                    <td
                      class=""
                    >
                      test1
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="1"
                  >
                    <td
                      class=""
                    >
                      2
                    </td>
                    <td
                      class=""
                    >
                      test2
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="2"
                  >
                    <td
                      class=""
                    >
                      3
                    </td>
                    <td
                      class=""
                    >
                      test3
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="3"
                  >
                    <td
                      class=""
                    >
                      4
                    </td>
                    <td
                      class=""
                    >
                      test4
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="4"
                  >
                    <td
                      class=""
                    >
                      5
                    </td>
                    <td
                      class=""
                    >
                      test5
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="5"
                  >
                    <td
                      class=""
                    >
                      6
                    </td>
                    <td
                      class=""
                    >
                      test6
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="6"
                  >
                    <td
                      class=""
                    >
                      7
                    </td>
                    <td
                      class=""
                    >
                      test7
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="7"
                  >
                    <td
                      class=""
                    >
                      8
                    </td>
                    <td
                      class=""
                    >
                      test8
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="8"
                  >
                    <td
                      class=""
                    >
                      9
                    </td>
                    <td
                      class=""
                    >
                      test9
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="9"
                  >
                    <td
                      class=""
                    >
                      10
                    </td>
                    <td
                      class=""
                    >
                      test10
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination pagination ant-table-pagination"
          unselectable="unselectable"
        >
          <li
            class="ant-pagination-total-text"
          >
            共2条
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-prev"
            title="Previous Page"
          >
            <a
              class="ant-pagination-item-link"
            >
              <i
                aria-label="icon: left"
                class="anticon anticon-left"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </i>
            </a>
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-next"
            title="Next Page"
          >
            <a
              class="ant-pagination-item-link"
            >
              <i
                aria-label="icon: right"
                class="anticon anticon-right"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                  />
                </svg>
              </i>
            </a>
          </li>
          <li
            class="ant-pagination-options"
          >
            <div
              class="ant-pagination-options-size-changer ant-select ant-select-enabled"
            >
              <div
                aria-autocomplete="list"
                aria-controls="test-uuid"
                aria-expanded="false"
                aria-haspopup="true"
                class="ant-select-selection
            ant-select-selection--single"
                role="combobox"
                tabindex="0"
              >
                <div
                  class="ant-select-selection__rendered"
                >
                  <div
                    class="ant-select-selection-selected-value"
                    style="display: block; opacity: 1;"
                    title="10 / page"
                  >
                    10 / page
                  </div>
                </div>
                <span
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <i
                    aria-label="icon: down"
                    class="anticon anticon-down ant-select-arrow-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </i>
                </span>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
